aws/vpn/lambda.tf

```diff
@@ -0,0 +1,105 @@
+resource "aws_iam_role" "lambda" {
+  name               = "iam_for_vpn_lambda"
+  assume_role_policy = data.aws_iam_policy_document.assume_role.json
+}
+
+data "aws_iam_policy_document" "assume_role" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_policy" "lambda_logging" {
+  name        = "vpn_lambda_logging"
+  path        = "/"
+  description = "IAM policy for logging from a lambda"
+  policy      = data.aws_iam_policy_document.lambda_logging.json
+}
+
+data "aws_iam_policy_document" "lambda_logging" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents"
+    ]
+
+    resources = [
+      "arn:aws:logs:*:*:*"
+    ]
+  }
+}
+
+resource "aws_iam_policy" "lambda_vpc" {
+  name        = "vpn_lambda_vpc"
+  path        = "/"
+  description = "IAM policy for allowing lambda to manage VPC"
+  policy      = data.aws_iam_policy_document.lambda_vpc.json
+}
+
+data "aws_iam_policy_document" "lambda_vpc" {
+  # checkov:skip=CKV_AWS_111: This is a development environment, no need to restrict the lambda permissions
+  # checkov:skip=CKV_AWS_356: This is a development environment, no need to restrict the lambda permissions
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "ec2:DescribeClientVpnTargetNetworks",
+      "ec2:DisassociateClientVpnTargetNetwork",
+      "ec2:DescribeClientVpnConnections",
+      "ec2:DescribeClientVpnEndpoints"
+    ]
+
+    resources = [
+      "*"
+    ]
+  }
+}
```

Comment:
You can get rid of this policy since the Lambda isn't in a VPC.

Reply:
But doesn't the role need the policy in order to be able to modify resources of the VPC?

Reply:
Ah you are totally right.  My mistake.  I had read this as a policy needed to manage VPC resources and not VPN resources.